Sources/SwiftParser/Specifiers.swift

```diff
@@ -827,4 +872,14 @@ extension Parser {
     }
     return RawUnexpectedNodesSyntax(unexpected, arena: self.arena)
   }
+
+  mutating func virtualParse<T>(block: (inout Self) -> T) -> T {
+    let oldCurrentToken = currentToken
+    let oldLexemes = lexemes
+    defer {
+      currentToken = oldCurrentToken
+      lexemes = oldLexemes
+    }
+    return block(&self)
+  }
```

Comment:
Why do we need this? If you want to look ahead, you should be able to use a `Lookahead` by calling `lookahead()` on `Parser`. There are a number of places where we do that. 

But really, after parsing a misplaces `throws`, I would just check if the next token is a `(` which is not separated by any trivia from `throws` (ie. trailing trivia of `throws` is empty and leading trivia of `(` is empty). If it is, just continue parsing the thrown type.